src/main/java/org/owasp/benchmark/score/WriteTime.java

```diff
@@ -0,0 +1,389 @@
+				if (file.exists()) {
+					file.renameTo(new File("results/" + file.getName().replace(".zip", "-" + times + ".zip")));
+				}
+				break;
+		}
+	}
+
+	public void writeSonarResults() {
+
+		int page = 1;
+		int total = 1;
+		JSONArray issues = new JSONArray();
+		JSONObject json = null;
+
+		try {
+			while (issues.length() < total && page <= 20) {
+				json = new JSONObject(getSonarResults(SONAR_URL + "/api/issues/search?resolved=false&ps=500&p=" + page));
+				total = (int) json.get("total");
+
+				JSONArray issueSubset = json.getJSONArray("issues");
+				for (int i = 0; i < issueSubset.length(); i++) {
+					issues.put(issueSubset.get(i));
+				}
+				page++;
+			}
+
+			json.put("issues", issues);
+
+			String xml = XML.toString(json);
+			java.io.FileWriter fw = new java.io.FileWriter(SONAR_FILE);
+			fw.write(xml);
+			fw.close();
+		} catch (Exception e) {
+			System.out.println("There was an error while writing SonarQube results.");
+		}
+	}
+    
+    public static String getSonarVersion(String sonarURL) {
+        return getHttpResponse(sonarURL, "There was an error trying to retrieve SonarQube version.");
+    }
+    
+    public static String getSonarResults(String sonarURL){
+        return getHttpResponse(sonarURL, "There was an error trying to retrieve SonarQube results.");
+    }
+    
+	public static String getHttpResponse(String url, String errorMessege) {
+		StringBuffer response = new StringBuffer();
+		try {
+			URL obj = new URL(url);
+			HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+			con.setRequestMethod("GET");
+			con.setRequestProperty("User-Agent", USER_AGENT);
+
+			BufferedReader in = new BufferedReader(new InputStreamReader(
+					con.getInputStream()));
+			String inputLine;
+
+			while ((inputLine = in.readLine()) != null) {
+				response.append(inputLine);
+			}
+			in.close();
+		} catch (Exception e) {
+			System.out.println(errorMessege);
+		}
+		return response.toString();
+	}
+
+	public String getToolTime(String toolName) {
+		String[] results = new String[3];
+		String time = null;
+		List<String> lines = Utils.getLinesFromFile(Utils.getFileFromClasspath(
+				CSV_TIMES_FILE, this.getClass().getClassLoader()));
+		for (String i : lines) {
+			if (i.contains(toolName)) {
+				results = i.split(";");
+				time = results[2].replaceAll("\"", "");
+				// System.out.println(time.split("\\.")[0]);
+				return time.split("\\.")[0];
+			}
+		}
+		return "";
+	}
+
+	/**
+	 * Gets the current version of the Benchmark from the benchmark.properties file.
+	 * @return The version # (as a String). An empty string if its not defined in that file.
+	 * @throws Exception
+	 */
+	public String getbenchmarkVersion() { // throws Exception {
+		Properties benchMprops = new Properties();
+		try {
+			File propsFile = new File(this.getClass().getClassLoader().getResource(VERSION_FILE).toURI().getPath());
+			benchMprops.load(new FileInputStream(propsFile));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, e.g., when exception is thrown by Properties.load. Either a) close the object returned by *FileInputStream()* in a try-finally block or b) close the resource by declaring the object returned by *FileInputStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).